include/wx/translation.h

```diff
@@ -331,13 +331,13 @@ inline const wxString& wxGetTranslation(const char *str1,
 #if !defined(WXINTL_NO_GETTEXT_MACRO)
     #if !defined(_)
 #ifndef wxNO_IMPLICIT_WXSTRING_ENCODING
-        #define _(s)                 (s)
+        #define _(s)                 wxString(s)
```

Comment:
This looks correct but I find it a bit strange that we had never had problems with this before, i.e. AFAICS `_(s)` was always defined just as `s`, ever since it was added back in 88ac883a0d0.

Could you please explain what errors did this result in, i.e. why did you have to change it?

Reply:
While `_(s)` has been unchanged since the beginning, the function that it normally points to, `wxGetTranslation()`, has changed between 2.8 and 3.0, from returning `const wxChar*` to returning `const wxString &`.

It would be consistent that `_()`, `wxPLURAL()`, and `wxGETTEXT_IN_CONTEXT()` would all return a `wxString` also when `wxUSE_INTL` is off.

Reply:
The main (only?) reason for turning off `wxUSE_INTL` is to avoid extra code generation if you're not interested in i18n and `wxGetTranslation()` specifically does _not_ return `wxString` in this case in order to avoid constructing it unnecessarily. Also, there would be a difference (albeit a lesser one of course) between the 2 builds anyhow because the real `wxGetTranslation()` returns `const wxString&` and we couldn't return this in `wxUSE_INTL==0` case.

So I'd rather leave the header unchanged and fix the code to avoid assuming that `_()` or whatever returns a `wxString`. Especially because it seems to me -- and the current example proves it -- that concatenating translated strings with something else is rarely, if ever, the right thing to do anyhow.